lib/Locale/Country/Extra.pm

```diff
@@ -45,22 +46,32 @@ sub idd_from_code {
     return $self->_idd_codes->{$code};
 }
 
-sub code_from_phone {
-    my ( $self, $number ) = @_;
+sub get_valid_phone {
+    my ($self, $number) = @_;
+
+    return '' if $number =~ /^(111111|222222|333333|444444|555555|666666|777777|888888|999999|000000)/;
 
     $number =~ s/\D//g;    # Remove non-digits
     $number =~ s/^00//;    # Remove the leading '00'.
+    return $number;
+}
+
+sub code_from_phone {
+    my ( $self, $number ) = @_;
 
-    if ( $number !~
-/^(111111|222222|333333|444444|555555|666666|777777|888888|999999|000000)/
-      )
-    {
-        my %code_for_idd = reverse %{ $self->_idd_codes };
-        foreach my $iddcode ( sort { $b <=> $a } keys %code_for_idd ) {
-            if ( $number =~ /^$iddcode/ ) {
-                return lc $code_for_idd{$iddcode};
-            }
-        }
+    if (my $first = $self->codes_from_phone($number)) {
+        return lc ${$first}[0];
```

Comment:
Or simply
```suggestion
        return lc $first->[0];
```

though judging by the name of the variable, it is better to have just taken the array item upon initializing `$first`:

```perl
if (my $first = $self->codes_from_phone(number)->[0]) {
    return $first;
}